minifi/minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/service/MiNiFiConfigurationChangeListener.java

```diff
@@ -70,26 +79,23 @@ public void handleChange(InputStream configInputStream) throws ConfigurationChan
                 bufferedConfigOs.write(copyArray, 0, available);
             }
 
-            // Create an input stream to use for writing a config file as well as feeding to the config transformer
-            try (ByteArrayInputStream newConfigBais = new ByteArrayInputStream(bufferedConfigOs.toByteArray())) {
-                newConfigBais.mark(-1);
+            File swapConfigFile = bootstrapFileProvider.getSwapFile();
+            logger.info("Persisting old configuration to {}", swapConfigFile.getAbsolutePath());
 
-                File swapConfigFile = bootstrapFileProvider.getSwapFile();
-                logger.info("Persisting old configuration to {}", swapConfigFile.getAbsolutePath());
+            try (FileInputStream configFileInputStream = new FileInputStream(configFile)) {
+                Files.copy(configFileInputStream, swapConfigFile.toPath(), REPLACE_EXISTING);
+            }
 
-                try (FileInputStream configFileInputStream = new FileInputStream(configFile)) {
-                    Files.copy(configFileInputStream, swapConfigFile.toPath(), REPLACE_EXISTING);
-                }
+            persistBackNonFlowSectionsFromOriginalSchema(bufferedConfigOs.toByteArray(), bootstrapProperties, configFile);
+
+            // Create an input stream to feed to the config transformer
+            try (ByteArrayInputStream newConfigBais = new ByteArrayInputStream(IOUtils.toByteArray(new FileInputStream(configFile)))) {
+                newConfigBais.mark(-1);
 
                 try {
-                    logger.info("Persisting changes to {}", configFile.getAbsolutePath());
-                    saveFile(newConfigBais, configFile);
```

Comment:
This is unused now.

Reply:
Wanted to remove it but totally forgot, thanks. Removed it now.